### PR TITLE
Detect integers 6644 v12

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6644 and all subtickets
https://redmine.openinfosecfoundation.org/issues/6645
https://redmine.openinfosecfoundation.org/issues/6646
https://redmine.openinfosecfoundation.org/issues/6647
https://redmine.openinfosecfoundation.org/issues/6648
https://redmine.openinfosecfoundation.org/issues/6628

Describe changes:
- detect/integers: support hexadecimal notation for parsing
- detect/integers: add mode for negated range
- detect/integers: rust derive for enumerations
- detect/integers: keywords now accept bitmasks
- doc: detect/integers

https://github.com/OISF/suricata/pull/10222 with typo fixed in doc

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW1_files_sha256.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .app_layer.error.http.parser | 1108 | 724 | 65.34% | 

Pipeline 17729

**Comment by catenacyber** [comment]
Why is CI using libhtp 0.5.45 and not 0.5.x ?

**Comment by jasonish** [comment]
> Why is CI using libhtp 0.5.45 and not 0.5.x ?

CI uses `scripts/bundle.sh` which uses `requirements.txt` for repeatable builds in and out of CI.  `requirements.txt` hasn't been updated since the 7.0.0 release it looks like.

File: doc/userguide/rules/dhcp-keywords.rst [hunk 1]
```diff
@@ -6,6 +6,8 @@ dhcp.leasetime
 
 DHCP lease time (integer).
 
+dhcp.leasetime uses an :ref:`unsigned 64-bits integer <rules-integer-keywords>`.
```
Comment (jasonish) [inline doc/userguide/rules/dhcp-keywords.rst:9]: `64-bits` or `64-bit`?

File: rust/src/detect/mod.rs [hunk 2]
```diff
@@ -25,3 +25,46 @@ pub mod stream_size;
 pub mod uint;
 pub mod uri;
 pub mod requires;
+
+/// Enum trait that will be implemented on enums that
+/// derive StringEnum.
+pub trait Enum<T> {
```
Comment (jasonish) [inline rust/src/detect/mod.rs:31]: Would `EnumString` be a better name?

File: doc/userguide/rules/dhcp-keywords.rst (hunk 1 above)
Comment (catenacyber) [reply doc/userguide/rules/dhcp-keywords.rst:9]: Without an s if I believe 

> 16-bit integers are those that are 16 bits (2 octets) wide.

Fixing that

File: rust/src/detect/mod.rs (hunk 2 above)
Comment (catenacyber) [reply rust/src/detect/mod.rs:31]: indeed

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/10241

**Pull request closed without merging**